repose-aggregator/components/filters/openstack-identity-v3-filter/src/main/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3Handler.scala

```diff
@@ -87,6 +86,19 @@ class OpenStackIdentityV3Handler(identityConfig: OpenstackIdentityV3Config, iden
       response.addHeader(HttpHeaders.RETRY_AFTER, retry)
     }
 
+    implicit val catalogWrites = new Writes[ServiceForAuthenticationResponse] {
+      def writes(catalog: ServiceForAuthenticationResponse) = JsObject(Seq(
+        "endpoints" -> JsArray(catalog.endpoints.map(endpoint => JsObject(
+          Seq("id" -> JsString(endpoint.id)) ++
+            endpoint.name.map(name => Seq("name" -> JsString(name))).getOrElse(Seq()) ++
+            endpoint.interface.map(interface => Seq("interface" -> JsString(interface))).getOrElse(Seq()) ++
+            endpoint.region.map(region => Seq("region" -> JsString(region))).getOrElse(Seq()) ++
+            Seq("url" -> JsString(endpoint.url)) ++
+            endpoint.service_id.map(serviceId => Seq("service_id" -> JsString(serviceId))).getOrElse(Seq()))))) ++
+        catalog.id.map(id => Seq("id" -> JsString(id))).getOrElse(Seq()) ++
+        catalog.name.map(name => Seq("name" -> JsString(name))).getOrElse(Seq()))
+    }
```

Comment:
Not sure if this was one of the things you wanted to change, but you may be able to simplify this block using function construction. 

"Using class construction" on https://www.playframework.com/documentation/2.5.x/ScalaJson

Also, I always use `Collection.empty[Type]` over `Collection[Type]()`. It's a personal preference thing, but I find it to read more clearly and cut down on the sometimes obscene number of parenthesis you see nested in Scala.